wayland-scanner/src/client_gen.rs

```diff
@@ -81,6 +81,12 @@ fn generate_objects_for(interface: &Interface) -> TokenStream {
 
             impl std::cmp::Eq for #iface_name {}
 
+            impl PartialEq<Weak<#iface_name>> for #iface_name {
```

Comment:
Is there a reason there can't be a blanket implementation for this?

Reply:
Yeah, that conflicts with orphan rules.